### DISCUSSION TITLE
build qt

### DISCUSSION THREAD

**Discussion by zj951555793** [microsoft/vcpkg#31459] [General]
I have installed some libraries such as qt vtk pcl, and when I use vcpkg install qttools [qml] -- recurse

I think it's very unfriendly to rebuild the entire building system, which means that I wasted half a day building and another day rebuilding and producing these libraries

logs:
PS C:\Users\ruler9068> vcpkg install qttools[assistant,core,designer,linguist,qml]:x64-windows
Computing installation plan...
The following packages will be rebuilt:
  * pcl[apps,core,cuda,examples,libusb,opengl,openni2,pcap,qt,simulation,tools,visualization,vtk]:x64-windows -> 1.13.0#1
    qttools[assistant,core,designer,linguist,qml]:x64-windows -> 6.4.3
  * qttranslations[core,default-features]:x64-windows -> 6.4.3
  * vtk[core,opengl,qt]:x64-windows -> 9.2.0-pv5.11.0#5
Additional packages (*) will be modified to complete this operation.

**Comment by Neumann-A** [comment]
That is the price you pay for changing the feature set which forces a recompilation of a dependent library which subsequently requires a rebuild of all consumers. There is nothing unfriendly about it it is just how it is. 
If you use manifest mode your projects are better isolated however of for the price of more required space.
